src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfoOfT.WriteHelpers.cs

```diff
@@ -32,7 +32,7 @@ public partial class JsonTypeInfo<T>
                 // this avoids creating a WriteStack and calling into the converter infrastructure.
 
                 Debug.Assert(SerializeHandler != null);
-                Debug.Assert(Options.SerializerContext?.CanUseSerializationLogic == true);
+                Debug.Assert(Options.CanUseFastPathSerializationLogic);
```

Comment:
Are these changes from https://github.com/dotnet/runtime/pull/79828?

Reply:
Correct, I copied over the logic because it turned out that the same changes were needed here as well. Now that #79828 has been unblocked & merged I'll rebase the changes and hopefully that part of the diff will go away.